packages-next/fields/src/Implementation.ts

```diff
@@ -58,6 +60,7 @@ class Field<P extends string> {
     this.schemaDoc = schemaDoc;
     this.config = config;
     this.isRequired = !!isRequired;
+    this.isIndexed = !!isIndexed;
```

Comment:
Aren't you reading isIndexed from the adapter, not the implementation class? So this is doing nothing? but it still works because isIndexed is already set on the adapter class?